runtime/src/benchmarking/criterion.rs

```diff
@@ -0,0 +1,82 @@
+/// Criterion benchmark executor implementations.
+use crate::Runner;
+use criterion::async_executor::AsyncExecutor;
+use futures::Future;
+use std::any::Any;
+use std::cell::RefCell;
+use std::thread_local;
+
+thread_local! {
```

Comment:
Can move this to a util in `runtime` rather than putting in `benchmarking`.

Move `Executor` into `tokio`.